sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ProviderResourceTypeProxy.cs

```diff
@@ -0,0 +1,267 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+using ClientModel.Tests.ClientShared;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace System.ClientModel.Tests.Client.Models.ResourceManager.Resources
+{
+    public partial class ProviderResourceTypeProxy : IJsonModel<ProviderResourceType>
```

Comment:
A comment here explaining what the class is used for might help future readers/maintainers of this code.